### ISSUE TITLE
getting the below error while trying to add the user to a group

### ISSUE THREAD

**Issue by coolsambit** [SharePoint/PnP-JS-Core#439]
Thank you for reporting an issue, suggesting an enhancement, or asking a question. We appreciate your feedback - to help the team understand your
needs please complete the below template to ensure we have the details to help. Thanks!

#### Category

[ ] Bug

#### Version
v2.0.3

I thought this is the latest version.

#### Expected / Desired Behavior / Question
User getting added

#### Observed Behavior
 var item1 = "{'LoginName': 'i:0#.w|wb\wb469213 '}";
  $pnp.sp.web.siteGroups.getById(245).users.add(item1).then(function(d){
                                                                                           
                                                                                }).catch(function(error){
                                                                                                alert(error);
                                                                                });

Error : "ProcessHttpClientResponseException: Error making HttpClient request in queryable: [500] Internal Server Error
    at new t (https://kmdev.ifc.org/sites/pnp2/Style%20Library/JS/pnp.min.js:1:19200)
    at https://kmdev.ifc.org/sites/pnp2/Style%20Library/JS/pnp.min.js:1:12338"

#### Steps to Reproduce
_If you are reporting an issue please describe the steps to reproduce the bug in sufficient detail to allow testing. If you are making
a suggestion or asking a question delete this section._

Thanks!
Sambit

**Comment by patrick-rodgers** [comment]
Hi @coolsambit - can you try with just the login name portion, "'i:0#.w|wb\wb469213" as the argument? I also noted you have a space at the end of the login name string which may also cause issues. But for this I think just supplying the login name will work.

**Comment by patrick-rodgers** [comment]
Haven't heard back so going to close this as answered. Please reopen the ticket if you continue to have issues. and let us know. Thanks!

**Comment by coolsambit** [comment]
Thanks Patrick for the quick response. I did not get time to try it out before.
As suggested I tried both your suggestions.
1) Removed the space
2) wb\wb469213
In both cases it failed with the exact same error message as above.

**Comment by patrick-rodgers** [comment]
**Please reopen issue to continue the discussion**

Did you try the fully resolved login "i:0#.w|wb\wb469213" as suggested?

**Comment by coolsambit** [comment]
Yes. Tried all variations. 
I cannot be the only one getting this error. :(

**Comment by koltyakov** [comment]
How about "i:0#.w|wb\\wb469213" with `\\`?

**Comment by coolsambit** [comment]
Tried with \\ but no luck.

**Comment by koltyakov** [comment]
I use this way of adding users to a group:

```javascript
$pnp.sp.web
  .siteGroups.getById(17)
  .users.add('i:0#.w|wb\\wb469213')
  .then(console.log);
```

It works for me.

It's better to assure login name is correct by getting a list of users:

```javascript
$pnp.sp.web.siteUsers.get().then(users => {
   console.log(users.map(u => u.LoginName));
});
```

And get sure that there is `\\` in a string, if it manually formatted in a script.

**Comment by coolsambit** [comment]
Thanks alot for the tips and everyones involvement. 
There were couple of mistakes that I was doing.

i) 'wb\wb469213' user had a change of domain and was not valid. My Bad. 

I tried with another user and passed the details directly as suggested.
 .users.add('i:0#.w|wb\\wbxxxxx'). 
So, the \\\ trick worked and the user got added to the group.

ii) There was a double quote to the class. Not sure what I was thinking.
Made a small change. OOOPPPs...in JAVASCRIPT
Removed the double quote and passed "

> object.propertyName



var item1 = {'LoginName': 'i:0#.w|wb\wb469213 '};
$pnp.sp.web.siteGroups.getById(245)
                                       .users
                                       .add(item1.**LoginName**)
                                       .then(function(d){.....});
